pkg/sdn/plugin/subnets.go

```diff
@@ -95,11 +95,6 @@ func (master *OsdnMaster) deleteNode(nodeName string) error {
 	if err != nil {
 		return fmt.Errorf("Error fetching subnet for node %q for deletion: %v", nodeName, err)
 	}
-	_, ipnet, err := net.ParseCIDR(sub.Subnet)
-	if err != nil {
-		return fmt.Errorf("Error parsing subnet %q for node %q for deletion: %v", sub.Subnet, nodeName, err)
-	}
-	master.subnetAllocator.ReleaseNetwork(ipnet)
```

Comment:
hostsubnet is an object by itself. The actual subnet it refers to just needs to be checked back into the pool and its release can be delayed forever no problem (only that we may run out of subnets in the cidr it is super slow). 

If the node is reborn it will be given a new subnet with a new hostsubnet object. And if the original subnet is still not released (because the event hostsubnet:delete was not delivered), it will just never be able to get its old lease back. There was never a guarantee of that anyway.